src/main/java/org/jitsi/videobridge/simulcast/sendmodes/RewritingSendMode.java

```diff
@@ -115,42 +118,56 @@ public void receive(SimulcastStream simStream, boolean urgent)
         if (logger.isDebugEnabled())
         {
             logger.debug("order-" + simStream.getOrder()
-                    + " is the target (urgent:" + urgent + ") from " +
+                    + " is the target from " +
                     getSimulcastSender().getSimulcastReceiver()
                     .getSimulcastEngine()
                     .getVideoChannel().getEndpoint().getID() + ".");
         }
 
         simStream.askForKeyframe();
-        if (urgent || current == null)
+        if (current == null)
         {
-            weakCurrent = new WeakReference<>(simStream);
+            this.state
+                = new State(new WeakReference<>(simStream), oldState.weakNext);
         }
         else
         {
-            weakNext = new WeakReference<>(simStream);
+            this.state
+                = new State(oldState.weakCurrent, new WeakReference<>(simStream));
         }
     }
 
     /**
-     * Gets the <tt>SimulcastStream</tt> that is currently being received.
-     *
-     * @return
+     * A simple class that holds the state of a {@RewritingSendMode}.
      */
-    public SimulcastStream getCurrent()
+    static class State
     {
-        WeakReference<SimulcastStream> wr = this.weakCurrent;
-        return (wr != null) ? wr.get() : null;
-    }
+        /**
+         * Ctor.
+         *
+         * @param weakCurrent A <tt>WeakReference</tt> to the
+         * <tt>SimulcastStream</tt> that is currently being received.
+         * @param weakNext A <tt>WeakReference</tt> to the
+         * <tt>SimulcastStream</tt> that is currently being received.
+         */
+        public State(
+            WeakReference<SimulcastStream> weakCurrent,
+            WeakReference<SimulcastStream> weakNext)
+        {
+            this.weakCurrent = weakCurrent;
+            this.weakNext = weakNext;
+        }
 
-    /**
-     * Gets the <tt>SimulcastStream</tt> that was previously being received.
-     *
-     * @return
-     */
-    public SimulcastStream getNext()
-    {
-        WeakReference<SimulcastStream> wr = this.weakNext;
-        return (wr != null) ? wr.get() : null;
+        /**
+         * A <tt>WeakReference</tt> to the <tt>SimulcastStream</tt> that is
+         * currently being received.
+         */
+        private final WeakReference<SimulcastStream> weakCurrent;
```

Comment:
Since this is moved to a class of it's own, can we hide the WeakReference complexity with a getter and setter?